feconf.py

```diff
@@ -20,18 +20,24 @@
 import datetime
 import os
 
+
 # Whether to unconditionally log info messages.
 DEBUG = False
 
 # The platform for the storage backend. This is used in the model-switching
 # code in core/platform.
 PLATFORM = 'gae'
 
+# This should be string comparison, since all environment variables
+# are converted to string
+IS_MINIFIED = os.environ.get('MINIFICATION') == 'True'
+
 # Whether we should serve the development or production experience.
 if PLATFORM == 'gae':
     DEV_MODE = (
-        not os.environ.get('SERVER_SOFTWARE')
-        or os.environ['SERVER_SOFTWARE'].startswith('Development'))
+        (not os.environ.get('SERVER_SOFTWARE')
+         or os.environ['SERVER_SOFTWARE'].startswith('Development'))
+        and not IS_MINIFIED)
```

Comment:
I think I do lean toward splitting up DEV_MODE into MINIFICATION and everything else called 'DEV_MODE' to keep consistency, but I'm worried this PR might be a bit too much to chew. @MAKOSCAFEE do you have a good idea how hard it would be split DEV_MODE truly into two pieces: one MINIFICATION flag for all things MINIFICATION and the existing DEV_MODE for everything else?